psydac/linalg/topetsc.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+
+from psydac.linalg.block import BlockVectorSpace, BlockVector, BlockMatrix
+from psydac.linalg.stencil import StencilVectorSpace, StencilVector
+from scipy.sparse import coo_matrix, bmat
+
+from mpi4py import MPI
+
+def get_npts( space ):
+
+    if isinstance(space, StencilVectorSpace):
+        npts = np.product(space.npts)
+    elif isinstance(space, BlockVectorSpace):
+        npts = sum([get_npts(s) for s in space.spaces])
+    else:
+        raise TypeError("Expected StencilVectorSpace or BlockVectorSpace, found instead {}".format(type(space)))
+
+    return npts
+
+def vec_tocoo( vec ):
```

Comment:
Please add a docstring

Reply:
done in [68937d4](https://github.com/pyccel/psydac/pull/244/commits/68937d4ffbcceec954f6f2e2bef2287f88f62853)